### PR TITLE
Fix log axis range below 1

### LINKED ISSUE #14352: Log Axis and Range1d < 1 failing
### Software versions

Python version        :  3.12.3 | packaged by conda-forge | (main, Apr 15 2024, 18:20:11) [MSC v.1938 64 bit (AMD64)]
IPython version       :  8.29.0
Tornado version       :  6.4.1
Bokeh version         :  3.4.1
BokehJS static path   :  C:\ProgramData\Anaconda3\envs\geo_env312v1\Lib\site-packages\bokeh\server\static
node.js version       :  v22.11.0
npm version           :  10.9.0
jupyter_bokeh version :  (not installed)
Operating system      :  Windows-10-10.0.19045-SP0

### Browser name and version

_No response_

### Jupyter notebook / Jupyter Lab version

_No response_

### Expected behavior

I'm using an "outdated" version of bokeh (3.4.1), but seems like a regression of this old issue, from a long time ago and I suspect it's still a problem in latest release -->  https://github.com/bokeh/bokeh/issues/5389

I'm supplying a spec'd x range whose max value is less than one on a log scaled axis.

### Observed behavior

Figure fails to render in browser.

I get this printout JS side:

Image

### Example code

```Python
from bokeh.plotting import figure,show
from bokeh.models import Range1d

f = figure(x_axis_type='log'
            ,x_range=Range1d(start=1e-8,end=1e-6)
           # ,x_range=[1e-8,1e-6] #same result
           # ,x_range = Range1d(start=1e-8,end=3) #works
           )
show(f)
```

### Stack traceback or browser console output

_No response_

### Screenshots

_No response_

### PR DESCRIPTION
**Pull request by muendlein**
- [x] issues: fixes #14352
- [x] tests added / passed

### CODE REVIEW

**Comment by muendlein** [comment]
@bryevdv The bug fix seems to affect a lot baseline image while still being technically correct. This is essentially caused by the interval spacing between the major ticks which allows for multiple correct solutions.
Do you have any recommendation how proceed with that, e.g. override those baseline images or enforce the old behavior?

Old:
image


New:

image

Update: Fixed by using the first possible tick from the left.

**Comment by mosc9575** [comment]
Could you please add a test with a negative log axis to check if your PR also solves #14478?

**Comment by muendlein** [comment]
@mosc9575 I just checked, this PR won't solve #14478. Supporting negative log axis will probably require some more work as the whole range must be either completely negative or positive.

File: bokehjs/src/lib/models/tickers/log_ticker.ts [hunk 1]
```diff
@@ -61,20 +61,24 @@ export class LogTicker extends AdaptiveTicker {
         }
       }
     } else {
-      const startlog = Math.ceil(log_low * 0.999999)
-      const endlog = Math.floor(log_high * 1.000001)
+      const sign_log_low = Math.sign(log_low)
+      const sign_log_high = Math.sign(log_high)
+      const d = 0.000001
+      const startlog = Math.ceil(log_low * (1.0-sign_log_low*d))
+      const endlog = Math.floor(log_high * (1.0+sign_log_high*d))
```
Comment (bryevdv) [inline bokehjs/src/lib/models/tickers/log_ticker.ts]: Just a suggestion, but I would find something like this a little easier to follow, re: intention
```suggestion
      const d = 0.000001
      const low_pad = 1.0 - Math.sign(log_low)*d
      const high_pad = 1.0 + Math.sign(log_high)*d 
      const startlog = Math.ceil(log_low * low_pad)
      const endlog = Math.floor(log_high * high_pad)
```

File: bokehjs/src/lib/models/tickers/log_ticker.ts [hunk 2]
```diff
@@ -61,20 +61,24 @@ export class LogTicker extends AdaptiveTicker {
...
       const interval = Math.ceil((endlog - startlog) / 9.0)
 
-      ticks = range(startlog-1, endlog+1, interval)
-        .map((i) => base**i)
+      let tick_options = range(startlog-1, endlog+1, 1).map((i) => base**i)
+      tick_options = tick_options.filter((tick) => data_low <= tick && tick <= data_high)
```
Comment (bryevdv) [inline bokehjs/src/lib/models/tickers/log_ticker.ts]: Maybe
```suggestion
      const base_tick_options = range(startlog-1, endlog+1, 1).map((i) => base**i)
      const tick_options = base_tick_options.filter((tick) => data_low <= tick && tick <= data_high)
```

**Review by bryevdv — APPROVED** [review]
